app/models/akka/RequesterBehavior.scala

```diff
@@ -161,4 +274,29 @@ trait RequesterBehavior { me: AbstractDSLinkActor =>
 
     (resolveTargetByPath orElse resolveUnsubscribeTarget orElse resolveCloseTarget)(request)
   }
+}
+
+case class PathAndQos(path:String, qos:QoS.Level)
+
+object QoS {
+  sealed abstract class Level(val index:Int){
+    def <(other:Level) = index < other.index
+    def >(other:Level) = index > other.index
+    def >=(other:Level) = index >= other.index
+    def <=(other:Level) = index <= other.index
+  }
+
+  case object Default extends Level(0)
+  case object Queued extends Level(1)
+  case object Durable extends Level(2)
+  case object DurableAndPersist extends Level(3)
+
+
+  def apply(level:Int): QoS.Level = level match {
+    case 0 => Default
+    case 1 => Queued
+    case 2 => Durable
+    case 3 => DurableAndPersist
+    case  _ => throw new RuntimeException("unsupported QoS level")
```

Comment:
Very minor, just for forensics, let's output the bad value to the log: `RuntimeException(s"unsupported QoS level: $level")`

Reply:
done